### PR TITLE
refactor LevelBadge to remove redundant usages of useState &useEffect

### PR DESCRIPTION
**Pull request by hjaintech**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

LevelBadge component was using `useState` and `useEffect` hook when neither of them were needed. Refactored out usage of both the hooks in this PR

Fixes https://github.com/reactplay/react-play/issues/1225

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output

### CODE REVIEW

**Comment by atapas** [comment]
@hjaintech changes look good. We need an issue to merge the PR. Could you please create one and update the description with the issue number? We will merge it after that.

**Review by atapas — APPROVED** [review]
Looks good.. Waiting for isue to be created to merge. @hjaintech

**Comment by hjaintech** [comment]
> Looks good.. Waiting for isue to be created to merge. @hjaintech

Done @atapas ✅

**Pull request merged**
